src/nvim/if_cscope.c

```diff
@@ -1772,11 +1777,14 @@ static int cs_read_prompt(int i)
       if (bufpos < maxlen - 1 && vim_isprintc(ch)) {
         // lazy buffer allocation
         if (buf == NULL) {
-          buf = xmalloc(maxlen);
+          assert(maxlen >= 0);
+          buf = xmalloc((size_t)maxlen);
         }
         {
           /* append character to the message */
-          buf[bufpos++] = ch;
+          assert(ch >= 0);
+          assert(ch <= CHAR_MAX);
```

Comment:
I am very new to neovim. So my comments might not be best practice for neovim.
Does neovim accepts following assert style? 
I would rather go with single assert like below
assert( (ch >= CHAR_MIN && ch <= CHAR_MAX) && "ch is not in range of char");